### PR TITLE
Add command line script

### PR DESCRIPTION
**Pull request by Ivendrakeo**
# Date

Initial PR Date: 3-22-21

Due Date: 3-25-21

# Reviewers

@DeclanBruce 
@rockboynton 

# Description

Adds a command line script to allow the user to cleanly interface with the program manager. They can call the script by `$hobby-hub` anywhere.

The options are as follows:
[-h|--help]                     usage instructions. Help output.
[-r|--restart]                  restarts the currently running programs.
[-s|--stop]                     kills the currently running and paused programs. 
[-p|--pause]                    pauses the currently running programs.
[-c|--continue]                 resumes currently paused programs.
[-l|--list]                     lists currently running programs.
[-t|--transfer] program_file    transfers program to staging directory.
[-z|--remove] program_file      stop and remove provided program.
[-d|--debug] program_file       temporarily runs provided program file.
[-dr|--dbgResume]               terminates debug and resumes state.

*if any of these descriptions are unclear, let me know and I can change them.

View the wikipage on usage here: [interfacing with the program manager usage](https://github.com/Senior-Design-0x07/the-hobby-hub/wiki/Interfacing-With-The-Program-Manager----Utility)

Also, this PR includes revisions to the program manager to allow it to kick off additionally programs after starting the intial directory. Adding additional programs to the program manager can be done using the associated command or by the following:

`python3 /etc/hobby-hub/program_manager.py /etc/hobby-hub/test_programs/$1 >> /etc/hobby-hub/out/running_programs.txt`

Lastly, we had to modify the  sudoers file to allow dev to issue these commands.

# Project Issue

[Interface to allow user to interact with the program manager #42](github.com/Senior-Design-0x07/the-hobby-hub/issues/42)

# Test Results

I have done fairly exhaustive testing as debian and everything works, however further testing should be done as dev.

awaiting results from testing with peers.

### CODE REVIEW

File: files/src/hobby-hub [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+    echo "[-p|--pause]                    pauses the currently running programs."
+    echo "[-c|--continue]                 resumes currently paused programs."
+    echo "[-l|--list]                     lists currently running programs."
+    echo "[-t|--transfer] program_file    transfers program to staging directory."
+    echo "[-z|--remove] program_file      stop and remove provided program."
+    echo "[-d|--debug] program_file       temporarly runs provided program file."
+    echo "[-dr|--dbgResume]               terminates debug and resumes state."
+elif [ -s /etc/hobby-hub/out/debug_config.txt ]
+then
+# we're in a debug session
```
Comment (rockboynton) [inline files/src/hobby-hub:191]: How does debugging work? Is there documentation anywhere?

File: files/system/commands/move_process.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+cp $1 /etc/hobby-hub/test_programs
```
Comment (rockboynton) [inline files/system/commands/move_process.sh]: newline

File: files/system/commands/remove_process.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+rm /etc/hobby-hub/test_programs/$1
```
Comment (rockboynton) [inline files/system/commands/remove_process.sh]: newline

File: files/system/commands/start_process.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+python3 /etc/hobby-hub/program_manager.py /etc/hobby-hub/test_programs/$1 >> /etc/hobby-hub/out/running_programs.txt
```
Comment (rockboynton) [inline files/system/commands/start_process.sh]: newline

**Review by rockboynton — CHANGES_REQUESTED** [review]

File: files/src/hobby-hub (hunk 1 above)
Comment (Ivendrakeo) [reply files/src/hobby-hub:191]: debugging will pause all of the currently running programs and just run the program you provide without actually sending it to the staging directory. 

The idea is that if you want to test your code without actually sending it to the program manager and crashing everything.

Just like I suggested you make a wiki page detailing usage, I guess it's the same idea here. So i'll go ahead and make a usage guide and post it on the wiki to give you a better idea on what to test.

**Comment by Ivendrakeo** [comment]
I should note that if you want to use the transfer/debug/remove stuff you're probably going to need a test file that was either made as the dev user or permissions allow dev to edit it. ie: if you made the file as dev, you'll need to execute" `sudo chmod g+rw,o+rw GPIO_48_blink.py` as debian before moving to dev and running those commands.

File: files/src/hobby-hub (hunk 1 above)
Comment (Ivendrakeo) [reply files/src/hobby-hub:191]: https://github.com/Senior-Design-0x07/the-hobby-hub/wiki/Interfacing-With-The-Program-Manager----Utility

**Pull request merged**